x-pack/plugins/dashboard_enhanced/public/services/drilldowns/abstract_dashboard_drilldown/abstract_dashboard_drilldown.tsx

```diff
@@ -36,8 +38,12 @@ export interface Params {
 export abstract class AbstractDashboardDrilldown<Context extends object = object>
   implements Drilldown<Config, Context>
 {
-  constructor(protected readonly params: Params) {
-    this.ReactCollectConfig = (props) => <CollectConfigContainer {...props} params={this.params} />;
+  constructor(protected readonly params: Params, theme: ThemeServiceStart) {
+    this.ReactCollectConfig = (props) => (
+      <KibanaThemeProvider theme$={theme.theme$}>
```

Comment:
I think we have the same issue in `x-pack/plugins/lens/public/trigger_actions/open_in_discover_drilldown.tsx` and `x-pack/plugins/drilldowns/url_drilldown/public/lib/url_drilldown.tsx`


We introduced these uiToReact and reactToUi conversions because we wanted our APIs to be React agnostic, but I wonder if in 2 years this opinion has changed it is better to just use plain React. 

Then it a way to fix this and simplify is to remove intermediate non-react conversion steps. 

@vadimkibana, wdyt? 


Of course we can also fix this theming issue by re-wrapping each instance in `KibanaThemeProvider` or do it inside `reactToUiComponent` helper and pass theme.theme$ there

Reply:
> We introduced these uiToReact and reactToUi conversions because we wanted our APIs to be React agnostic, but I wonder if in 2 years this opinion has changed it is better to just use plain React.

+1 to remove these conversions and just use plain react.

Reply:
+1 to just use plain react.

Reply:
+1 as well for using React directly without breaking the React context. Historically at Kibana there was a trend to abstract everything away from a rendering library, e.g. keep Kibana agnostic to the view implementation. But in general, I believe, we are reversing this to simplify code everywhere.

@nreese maybe in `ui_actions` where `UiComponent` is required you could also allow for `React.FunctionalComponent` as well, and then we don't need this conversion and drilldown configurator can check if the thing is already a React component, then just use it directly.